singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java

```diff
@@ -271,7 +288,11 @@ private void includeAuditMessageInBatch(List<ProducerRecord<byte[], byte[]>> mes
 
       ProducerRecord<byte[], byte[]> keyedMessage;
       for (int i = 0; i < validPartitions.size(); i++) {
-        buckets.add(new ArrayList<>());
+        // for each partitionId, there is a corresponding bucket in buckets and a corresponding
+        // headersMap in mapOfHeadersMaps.
+        int partitionId = validPartitions.get(i).partition();
+        buckets.put(partitionId, new ArrayList<>());
+        mapOfHeadersMaps.put(partitionId ,new HashMap<Integer, LoggingAuditHeaders>());
```

Comment:
nit: simplify as `        mapOfHeadersMaps.put(partitionId ,new HashMap<>());`  as `Integer, LoggingAuditHeaders` part is really not needed

Reply:
made the change in the update of the PR.  let java compiler to infer the type at the runtime